### PR TITLE
Fix bbox/tracking 2D/3D list confidence

### PR DESCRIPTION
**Pull request by iliiliiliili**
- Fix `BoundingBoxList`, `TrackingAnnotationList` `BoundingBoxList3D`, `TrackingAnnotationList3D` confidence warnings on empty lists #362 
- Fix undefined `image_id` and `segmentation` for coco `BoundingBoxList`

### CODE REVIEW

File: src/opendr/engine/target.py [hunk 1]
```diff
@@ -437,11 +437,13 @@ class BoundingBoxList(Target):
     """
     def __init__(
         self,
-        boxes,
+        boxes=[],
```
Comment (tsampazk) [inline src/opendr/engine/target.py:440]: Default value shouldn't be a mutable value, i would suggest defaulting to `None` and inside the method check for `None` and assign empty list if so.

**Review by tsampazk — COMMENTED** [review]
Thanks Illia for the fixes, changes look good, i have just one comment.

File: src/opendr/engine/target.py (hunk 1 above)
Comment (iliiliiliili) [reply src/opendr/engine/target.py]: Fixed

**Comment by iliiliiliili** [comment]
Some strange errors occur across different tests with no module found, such as:
`ModuleNotFoundError: No module named 'cv2'`

**Comment by tsampazk** [comment]
> Some strange errors occur across different tests with no module found, such as: `ModuleNotFoundError: No module named 'cv2'`

There seems to be some wider issue at play here, i don't think the errors in the tests are related to the changes in this PR.

**Comment by ad-daniel** [comment]
I've re-run the PR and the problems seem to have gone. Likely a CI hiccup

**Review by tsampazk — APPROVED** [review]
Thanks!

**Pull request merged**